pkg/stub/imagestreams.go

```diff
@@ -222,29 +219,32 @@ func (h *Handler) updateDockerPullSpec(oldies []string, imagestream *imagev1.Ima
 	case "jenkins-agent-maven":
 		return
 	}
-	if len(opcfg.Spec.SamplesRegistry) > 0 {
-		logrus.Debugf("updateDockerPullSpec stream %s has repo %s", imagestream.Name, imagestream.Spec.DockerImageRepository)
-		// don't mess with deprecated field unless it is actually set with something
-		if len(imagestream.Spec.DockerImageRepository) > 0 &&
-			!strings.HasPrefix(imagestream.Spec.DockerImageRepository, opcfg.Spec.SamplesRegistry) {
-			// if not one of our 4 defaults ...
-			imagestream.Spec.DockerImageRepository = h.coreUpdateDockerPullSpec(imagestream.Spec.DockerImageRepository,
-				opcfg.Spec.SamplesRegistry,
-				oldies)
-		}
 
-		for _, tagref := range imagestream.Spec.Tags {
-			logrus.Debugf("updateDockerPullSpec stream %s and tag %s has from %#v", imagestream.Name, tagref.Name, tagref.From)
-			if tagref.From != nil {
-				switch tagref.From.Kind {
-				// ImageStreamTag and ImageStreamImage will ultimately point to a DockerImage From object reference
-				// we are only updating the actual registry pull specs
-				case "DockerImage":
-					if !strings.HasPrefix(tagref.From.Name, opcfg.Spec.SamplesRegistry) {
-						tagref.From.Name = h.coreUpdateDockerPullSpec(tagref.From.Name,
-							opcfg.Spec.SamplesRegistry,
-							oldies)
-					}
+	if len(opcfg.Spec.SamplesRegistry) == 0 {
+		return
+	}
+
+	logrus.Debugf("updateDockerPullSpec stream %s has repo %s", imagestream.Name, imagestream.Spec.DockerImageRepository)
+	// don't mess with deprecated field unless it is actually set with something
+	if len(imagestream.Spec.DockerImageRepository) > 0 &&
+		!strings.HasPrefix(imagestream.Spec.DockerImageRepository, opcfg.Spec.SamplesRegistry) {
+		// if not one of our 4 defaults ...
+		imagestream.Spec.DockerImageRepository = h.coreUpdateDockerPullSpec(imagestream.Spec.DockerImageRepository,
+			opcfg.Spec.SamplesRegistry,
+			oldies)
+	}
+
+	for _, tagref := range imagestream.Spec.Tags {
+		logrus.Debugf("updateDockerPullSpec stream %s and tag %s has from %#v", imagestream.Name, tagref.Name, tagref.From)
+		if tagref.From != nil {
+			switch tagref.From.Kind {
+			// ImageStreamTag and ImageStreamImage will ultimately point to a DockerImage From object reference
+			// we are only updating the actual registry pull specs
+			case "DockerImage":
+				if !strings.HasPrefix(tagref.From.Name, opcfg.Spec.SamplesRegistry) {
+					tagref.From.Name = h.coreUpdateDockerPullSpec(tagref.From.Name,
+						opcfg.Spec.SamplesRegistry,
+						oldies)
```

Comment:
future improvement:  parse the pullspec and replace the registryhostname section, rather than searching+replacing specific strings.  That would future proof this against imagestreams that reference additional registries beyond what "oldies" knows about.

same for the logic above.

Reply:
I'll make a note to add a TODO in perf/fix PR